runtime/cudaq/platform/qpu.h

```diff
@@ -50,7 +52,17 @@ class QPU : public registry::RegisteredType<QPU> {
   /// observation and perform state-preparation circuit measurement
   /// based on the `spin_op` terms.
   void handleObservation(ExecutionContext *localContext) {
-    if (localContext && localContext->name == "observe") {
+    // The reason for the 2 if checks is simply to do a flushGateQueue() before
+    // initiating the trace.
+    bool execute = localContext && localContext->name == "observe";
+    if (execute) {
+      cudaq::ScopedTrace trace(cudaq::TIMING_OBSERVE,
+                               "handleObservation flushGateQueue()");
+      getExecutionManager()->flushGateQueue();
+    }
```

Comment:
@1tnguyen - I think that since I am adding this code here, that the new `ScopedTrace` and `flushGateQueue` you added to `runtime/nvqir/custatevec/CuStateVecCircuitSimulator.cu / observe()` is no longer needed. Do you agree?

Reply:
Yes, I do. 

I missed the GitHub notification for this comment. Sorry about that!

Reply:
Updated in 8634b4e2c